hazelcast/src/main/java/com/hazelcast/ClientMain.java

```diff
@@ -11,8 +11,8 @@
 public class ClientMain {
 
     public static void main(String[] args) {
-        System.setProperty("hazelcast.tpc.enabled", "true");
-        System.setProperty("hazelcast.tpc.eventloop.count", "" + Runtime.getRuntime().availableProcessors());
+        System.setProperty("hz.alto.enabled", "true");
+        System.setProperty("hz.alto.eventloop-count", "" + Runtime.getRuntime().availableProcessors());
```

Comment:
@pveentjer did you see this? The defaults are configurable via system properties like this. Is this not enough?

Reply:
I think the only problem with the config overrides is that they won't apply when you use programmatic API for configuration. Also, according to the reference manual, when you override a value in a block, you have to override every other value in the block otherwise the rest is reset to their defaults https://docs.hazelcast.com/hazelcast/5.2/configuration/configuring-with-system-properties

But, I did not understand why we have to make everything configurable through system properties. What is wrong with XML/YAML or programmatic configuration? @pveentjer what is the reasoning behind that?

Reply:
Not everything needs to be configured using system properties. But the above 2 properties are the 2 most fundamental ones to have configurable. You don't want to modify the application context for everything. It makes testing/production and problem-solving a lot easier.

Reply:
https://github.com/hazelcast/hazelcast/pull/23319/commits/009609eec0df96ba2a9327217aa40ce9d50c417b